controllers/k6_controller.go

```diff
@@ -93,5 +100,27 @@ func (r *K6Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&v1alpha1.K6{}).
 		Owns(&batchv1.Job{}).
+		Watches(&source.Kind{Type: &v1.Pod{}},
+			&handler.EnqueueRequestsFromMapFunc{ToRequests: handler.ToRequestsFunc(
+				func(object handler.MapObject) []reconcile.Request {
+					pod := object.Object.(*v1.Pod)
+					k6CrName, ok := pod.GetLabels()[k6CrLabelName]
+					if !ok {
+						return nil
+					}
+					return []reconcile.Request{
+						{NamespacedName: types.NamespacedName{
+							Name:      k6CrName,
+							Namespace: object.Meta.GetNamespace(),
+						}}}
+				})},
+			builder.WithPredicates(predicate.NewPredicateFuncs(func(meta metav1.Object, object runtime.Object) bool {
+				pod := object.(*v1.Pod)
+				_, ok := pod.GetLabels()[k6CrLabelName]
+				if !ok {
+					return false
+				}
+				return true
+			}))).
```

Comment:
This part "reads" much better now! Clear and concise :) We might want to move it out at some point in the future but right now it seems just right. And an added bonus that it is all in one place and we see that only Pods are supposed to be among objects so even type assertion without error looks "safer".
Thanks for this :+1: